src/Subscriptions/Subscription.php

```diff
@@ -413,14 +426,25 @@ public function is_first_payment( Payment $payment ) {
 	 * @return DateTimeImmutable|null
 	 */
 	public function get_next_payment_date() {
-		// Get current phase.
-		$phase = $this->get_current_phase();
+		return $this->next_payment_date;
+	}
 
-		if ( null === $phase ) {
-			return null;
+	/**
+	 * Set the next payment date of this subscription.
+	 *
+	 * @param DateTimeInterface $date Date.
+	 * @return void
+	 */
+	public function set_next_payment_date( $date ) {
+		$end_date = $this->get_end_date();
+
+		if ( null !== $end_date && $date >= $end_date ) {
+			$this->next_payment_date = null;
+
+			return;
 		}
```

Comment:
What if is a subscription is 'On hold' and reactivated by an admin on/after the subscription end date to for example get paid for the last period? Admin will be unable to update the next payment date.

Reply:
I believe `$phase->get_next_date()` already returned `null` when `( null !== $end_date && $date >= $end_date )` is `true`, i have opened a new issue for this: https://github.com/pronamic/wp-pay-core/issues/62.